op-challenger/fault/agent.go

```diff
@@ -50,6 +53,18 @@ func (a *Agent) Act() error {
 	return nil
 }
 
+// resolveAndExit resolves the game if it is in a terminal state
+// and returns true if the game resolves successfully.
+func (a *Agent) resolveAndExit(ctx context.Context) bool {
```

Comment:
maybe this should be `tryResolve`? It doesn't actually exit so not sure that should be part of the name and it doesn't always resolve so that should be reflected in the name somehow.